tests/5.Transaction.js

```diff
@@ -97,6 +98,40 @@ describe('Sign', () => {
     assert.equal(txnEnc.fee, 1000);
   });
 
+  it('should accept 0 fee', () => {
+    const o = {
+      from: '7ZUECA7HFLZTXENRV24SHLU4AVPUTMTTDUFUBNBD64C73F3UHRTHAIOF6Q',
+      to: '7ZUECA7HFLZTXENRV24SHLU4AVPUTMTTDUFUBNBD64C73F3UHRTHAIOF6Q',
+      fee: 0,
+      flatFee: true,
+      amount: 847,
+      firstRound: 51,
+      lastRound: 61,
+      genesisHash: 'JgsgCaCTqIaLeVhyL6XlRu3n7Rfk2FxMeK+wRSaQ7dI=',
+      note: new Uint8Array([123, 12, 200]),
+    };
+    const txn = new algosdk.Transaction(o);
+    assert.equal(txn.fee, 0);
+  });
+
+  it('should accept lower than min fee', () => {
+    const o = {
+      from: '7ZUECA7HFLZTXENRV24SHLU4AVPUTMTTDUFUBNBD64C73F3UHRTHAIOF6Q',
+      to: '7ZUECA7HFLZTXENRV24SHLU4AVPUTMTTDUFUBNBD64C73F3UHRTHAIOF6Q',
+      fee: 10,
+      flatFee: true,
+      amount: 847,
+      firstRound: 51,
+      lastRound: 61,
+      genesisHash: 'JgsgCaCTqIaLeVhyL6XlRu3n7Rfk2FxMeK+wRSaQ7dI=',
+      note: new Uint8Array([123, 12, 200]),
+    };
+    const txn = new algosdk.Transaction(o);
+    assert.equal(txn.fee, 10);
+    const txnEnc = txn.get_obj_for_encoding();
+    assert.equal(txnEnc.fee, 10);
+  });
+
```

Comment:
Could you add one more unit test where fee=0 and flatFee isn't mentioned to ensure that the txn will still have a fee of 1000?

Reply:
this case is covered after removing line 88.

Reply:
Good point, thanks